platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +102,168 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
+                LOGGER.debug("\t{} {}",
+                        filter.getProperty(Constants.SERVICE_RANKING),
+                        ServicePropertiesUtils.getStringProperty(filter,
+                                HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            }
+        }
     }
 
-}
+    private void initFilter(ServiceReference<SecurityFilter> securityFilterServiceReference,
+            FilterConfig filterConfig) throws ServletException {
+        final ServletContext servletContext;
+
+        if (filterConfig == null) {
+            LOGGER.warn(
+                    "DelegateServletFilter was initialized with a null filterConfig. Initializing SecurityFilter {} with null ServletContext",
+                    ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            servletContext = null;
+        } else {
+            servletContext = this.filterConfig.getServletContext();
+        }
+
+        getContext().getService(securityFilterServiceReference)
+                .init(new SecurityFilterInitFilterConfig(securityFilterServiceReference,
+                        servletContext));
+
+        LOGGER.debug("Initialized SecurityFilter {}",
+                ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+    }
+
+    public void removeSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+        sortedSecurityFilterServiceReferences.remove(securityFilterServiceReference);
+        getContext().getService(securityFilterServiceReference)
+                .destroy();
+    }
+
+    /**
+     * This inner class is used to instantiate a {@link FilterConfig} from a
+     * {@link SecurityFilter}'s service properties containing init params and the same
+     * {@link ServletContext} as the {@link DelegateServletFilter}. The {@link FilterConfig} is used
+     * to initialize the {@link SecurityFilter}.
+     * <p>
+     * The logic of this inner class is copied from the
+     * {@link org.ops4j.pax.web.extender.whiteboard} feature.
+     */
+    private class SecurityFilterInitFilterConfig implements FilterConfig {
+
+        private final ServiceReference<SecurityFilter> securityFilterServiceReference;
+
+        private final ServletContext servletContext;
+
+        private final Map<String, String> initParams;
+
+        SecurityFilterInitFilterConfig(
+                ServiceReference<SecurityFilter> securityFilterServiceReference,
+                ServletContext servletContext) {
+            this.securityFilterServiceReference = securityFilterServiceReference;
+            this.servletContext = servletContext;
+            initParams = createInitParams(securityFilterServiceReference);
```

Comment:
✏️ I don't see much point in passing the reference into this `createInitParams` method. If you were to make that method static, then it would make sense to pass this on the stack; however, as it's an instance method, it already has access to the state of the class.